src/utils.cpp

```diff
@@ -19,10 +19,18 @@ void read_moduli_from_csv( \
     mpz_init(n);
     int err = 0;
     bool zero = false;
+    cout << base << endl;
     while(true) {
         int id;
-        err = fscanf(file, "%d,", &id);
-        err = gmp_fscanf(file, "%Zx", n);
+        if (base == "base10"){
+            err = fscanf(file, "%d,", &id);
+            err = gmp_fscanf(file, "%Zd", n);
+            
+        } else  {
+            err = fscanf(file, "%d,", &id);
+            err = gmp_fscanf(file, "%Zx", n);
+        }
+        
```

Comment:
Probably better to put the first `fscanf` outside the `if`.

Reply:
(or even 
```
err = fscanf(file, "%d,", &id);
char * format = "%Zx";
if(base == 10) {
    format = "%Zd";
}
err = gmp_fscanf(file, format, n);
```

Reply:
Yes this is redundant

Reply:
I did not see your proposition, but I made sure that even if the user types -base16 it works and that the program only works for base 10 and 16